Modules/+Experiments/+AutoExperiment/@SpecSlowScan/analyze.m

```diff
@@ -191,12 +196,37 @@
         'amplitudes',NaN,...
         'widths',NaN,...
         'locations',NaN,...
+        'etas',NaN,...
         'background',NaN,...
         'index',NaN,... % Index into sites
         'redo',false,...
         'ignore',[]);     % indices of experiments in the fit
 end
 
+if preanalyze
+    err = [];
+    try
+        site_index = 1;
+        progbar = waitbar(site_index/n,'','Name','Analyzing all data','CreateCancelBtn','setappdata(gcbf,''canceling'',1)');
+        setappdata(progbar,'canceling',0);
+        update_all()
+        for curr_index=2:n
+            waitbar(curr_index/n,progbar,sprintf('Analyzing site %i/%i',curr_index,n));
+            drawnow limitrate
+            changeSite(curr_index);
+            if getappdata(progbar,'canceling')
+                break
+            end
+        end
+        changeSite(1); %end by going back to first site
+    catch err
```

Comment:
Perhaps try/catch should be within the for loop instead externally so a single error doesn't crash it?